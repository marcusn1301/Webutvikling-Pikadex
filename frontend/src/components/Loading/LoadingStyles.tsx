import styled from "styled-components";

export const LoadingOuter = styled.div`
  position: absolute;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const LoadingBall = styled.img`
    position: fixed;
    top: 50%;
    left: 50%;
    margin: -50px 0px 0px -50px;
    width: 100px;
    height: 100px;

  animation: App-logo-spin infinite 1s linear;

  @keyframes App-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`;

export const DotFlashing = styled.div`
  position: absolute;
  top: 65%;
  left: 50%;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #bd1808;
  color: #bd1808;
  animation: dotFlashing 1s infinite linear alternate;
  animation-delay: 0.5s;

  &::before,
  &::after {
    content: "";
    display: inline-block;
    position: absolute;
    top: 0;
  }

  &::before {
    left: -15px;
    width: 10px;
    height: 10px;
    border-radius: 5px;
    background-color: #bd1808;
    color: #bd1808;
    animation: dotFlashing 1s infinite alternate;
    animation-delay: 0s;
  }

  &::after {
    left: 15px;
    width: 10px;
    height: 10px;
    border-radius: 5px;
    background-color: #bd1808;
    color: #bd1808;
    animation: dotFlashing 1s infinite alternate;
    animation-delay: 1s;
  }

  @keyframes dotFlashing {
    0% {
      background-color: #bd1808;
    }
    50%,
    100% {
      background-color: #ebe6ff;
    }
  }
`;
