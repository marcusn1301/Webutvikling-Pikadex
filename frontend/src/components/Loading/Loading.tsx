import logo from "../../assets/pokeball.png";
import { DotFlashing, LoadingBall } from "./LoadingStyles";

const Loading = () => {
  return (
    <>
      <LoadingBall src={logo} />
      {/* <DotFlashing /> */}
    </>
  );
};

export default Loading;
