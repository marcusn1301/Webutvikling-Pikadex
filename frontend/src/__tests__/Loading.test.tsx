import React from 'react';
import {render, screen} from '@testing-library/react';
import renderer from "react-test-renderer";

import Loading from '../components/Loading/Loading';

test('Snapshot test for Navbar', () => {
    const container = renderer.create(<Loading />);
    expect(container).toMatchSnapshot();
});

