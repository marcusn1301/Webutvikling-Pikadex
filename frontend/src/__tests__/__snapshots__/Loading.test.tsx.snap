// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Navbar 1`] = `
<img
  className="sc-gswNZR iUDwuu"
  src="pokeball.png"
/>
`;
